tests/ui/check-cfg/well-known-values.rs

```diff
@@ -1,41 +1,90 @@
 //
 // check-pass
 // compile-flags: --check-cfg=cfg() -Z unstable-options
 
+#![feature(cfg_overflow_checks)]
+#![feature(cfg_target_has_atomic)]
+#![feature(cfg_target_has_atomic_equal_alignment)]
+#![feature(cfg_target_abi)]
+#![feature(cfg_relocation_model)]
+#![feature(cfg_sanitize)]
+#![feature(cfg_target_thread_local)]
+
+#[cfg(any(
+    doc = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    miri = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    unix = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    test = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    doctest = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    windows = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    proc_macro = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    panic = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    sanitize = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    relocation_model = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    debug_assertions = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    overflow_checks = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_thread_local = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_feature = "_UNEXPECTED_VALUE", // currently no values are checked
+    target_os = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_family = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_arch = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_endian = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_env = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_abi = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_vendor = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_pointer_width = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_has_atomic = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_has_atomic_load_store = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
+    target_has_atomic_equal_alignment = "_UNEXPECTED_VALUE"
+    //~^ WARN unexpected `cfg` condition value
+))]
+fn unexpected_values() {}
+
+// testing that we suggest `linux`
 #[cfg(target_os = "linuz")]
 //~^ WARNING unexpected `cfg` condition value
 fn target_os_linux_misspell() {}
 
 #[cfg(target_os = "linux")]
 fn target_os_linux() {}
 
-#[cfg(target_has_atomic = "0")]
-//~^ WARNING unexpected `cfg` condition value
-fn target_has_atomic_invalid() {}
-
 #[cfg(target_has_atomic = "8")]
 fn target_has_atomic() {}
```

Comment:
It's not clear to me why some of these remain, when they are now mentioned in the big list above? But some were removed.

Reply:
The linux_misspell one makes sense to remain, but I don't understand the rest.

Reply:
We still need to test that the combination of expected well known names and values do not produce a warning, but we do not need to test all of them, so I chose some that seemed relevant (with/without a value, both at the same time). It's very arbitrary and I'm happy to change it if desired.

Reply:
A comment explaining exactly that would be helpful :)

Reply:
Done. I've improved the comment at the start of the section.